langlib/langlib-test/src/test/resources/test-src/decimallib_test.bal

```diff
@@ -396,6 +396,24 @@ function testFromStringFunctionWithInvalidValues() {
         assertEquality("{ballerina/lang.decimal}NumberParsingError", a1.message());
         assertEquality("'string' value '' cannot be converted to 'decimal'", <string> checkpanic a1.detail()["message"]);
     }
+
+    a1 = decimal:fromString("1e-6143");
+    assertEquality(false, a1 is error);
+    assertEquality(1e-6143d, checkpanic a1);
+
+    a1 = trap decimal:fromString("9.999999999999999999999999999999999E6145");
```

Comment:
What happens with the hardcoded decimal values that are out of range?
```ballerina
    decimal x =  9.999999999999999999999999999999999E6145;
```
currently, it didn't throw any exceptions.

Reply:
It should result in compilation error and being addressed in https://github.com/ballerina-platform/ballerina-lang/pull/37823